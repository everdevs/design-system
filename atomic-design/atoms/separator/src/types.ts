import { PropsWithTheme } from "@evernest/theme";

export type SeparatorElement = HTMLDivElement;

export interface StyledSeparatorProps extends PropsWithTheme<SeparatorElement> {
	/** Add types */
}
