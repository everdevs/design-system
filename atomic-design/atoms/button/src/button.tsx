import { css } from "@emotion/core";
import styled from "@emotion/styled";
import { ButtonProps, Size, Variation } from "./types";

const Button = styled("button", {
	shouldForwardProp: (propName: string) => !["theme"].includes(propName),
})<ButtonProps>`
	cursor: pointer;
	border: 0;
	${({ theme, size, variation }) => css`
		box-shadow: ${theme.tokens.shadow.medium.css};
		height: ${size === Size.small ? "20px" : "40px"};
	`};
`;

Button.defaultProps = {
	type: "button",
	size: Size.medium,
	variation: Variation.default,
};

export { Button };
