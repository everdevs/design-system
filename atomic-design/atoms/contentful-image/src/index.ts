export * from "./contentful-image";
export * from "./utils";
export * from "./types";
