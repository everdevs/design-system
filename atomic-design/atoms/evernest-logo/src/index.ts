export * from "./evernest-logo";
export * from "./types";
