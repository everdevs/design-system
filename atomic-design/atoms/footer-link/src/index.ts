export * from "./footer-link";
export * from "./types";
