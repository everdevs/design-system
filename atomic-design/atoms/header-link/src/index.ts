export * from "./header-link";
export * from "./types";
