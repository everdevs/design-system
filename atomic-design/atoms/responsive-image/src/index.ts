export * from "./responsive-image";
export * from "./utils";
export * from "./types";
