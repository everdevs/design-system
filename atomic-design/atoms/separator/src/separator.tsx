import React from "react";
import styled from "@emotion/styled";
import { css } from "@emotion/react";
import { SeparatorElement, StyledSeparatorProps } from "./types";

export const StyledSeparator = styled.div<StyledSeparatorProps>`
	height: 1px;
	margin: var(--spacing-m) 0;
	${({ theme: { palette } }) => css`
		background: ${palette.brightGrey.css};
	`};
`;

export const Separator = React.forwardRef<SeparatorElement>((props, ref) => (
	<StyledSeparator {...props} ref={ref} />
));
