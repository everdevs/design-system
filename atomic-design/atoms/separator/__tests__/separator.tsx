import { createTree, withTheme } from "@evernest/dev-helpers";
import { theme } from "@evernest/theme";
import React from "react";
import { Separator, StyledSeparatorProps } from "../src";
import {mount} from "enzyme";

const ThemedSeparator = withTheme<StyledSeparatorProps>(Separator, theme);

test("Separator allows setting classname", () => {
	const className = "test";
	const wrapper = mount(<ThemedSeparator className={className} />);
	expect(wrapper).toHaveClassName(className);
});

test("Separator matches the snapshot", () => {
	const tree = createTree(<ThemedSeparator />);
	expect(tree).toMatchSnapshot();
});
