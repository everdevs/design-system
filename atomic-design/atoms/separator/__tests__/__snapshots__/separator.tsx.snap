// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator matches the snapshot 1`] = `
<div
  className="css-4nsmvf"
/>
`;
